### PR TITLE
Add dialog to select extentions in study creation

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/components/MultipleSelectionDialog/MultipleSelectionDialog.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                                control={
+                                    <Checkbox
+                                        checked={
+                                            selectedIds.length ===
+                                            options.length
+                                        }
+                                        indeterminate={
+                                            selectedIds.length !==
+                                                options.length &&
+                                            selectedIds.length !== 0
```
Comment (ghelloumi) [inline src/components/MultipleSelectionDialog/MultipleSelectionDialog.js:69]: I think that is better to do

selectedIds.length && selectedIds.length !== options.length 

(it is unnecessary to test selectedIds.length !== options.length if there is no selectedIds)

File: src/components/FlatParameters/FlatParameters.js [hunk 2]
```diff
@@ -268,6 +300,52 @@ export const FlatParameters = ({
                 );
             case 'STRING_LIST':
                 if (param.possibleValues) {
+                    const allOptions = sortPossibleValues(
+                        param.name,
+                        param.possibleValues
+                    ).map((v) => v.id);
```
Comment (ghelloumi) [inline src/components/FlatParameters/FlatParameters.js:306]: .map(({id}) => id);

File: src/components/FlatParameters/FlatParameters.js [hunk 3]
```diff
@@ -211,6 +231,18 @@ export const FlatParameters = ({
         return value?.replace(',', '.') || '';
     };
 
+    const getStringListValue = (allValues, selectValues) => {
+        if (!Array.isArray(selectValues) || selectValues?.length === 0) {
```
Comment (ghelloumi) [inline src/components/FlatParameters/FlatParameters.js:235]: can selectedValues be with type other than array if defined?
if not you can simply use
if (!selectValues || !selectValues.length)

**Review by ghelloumi — APPROVED** [review]
code ok
test ok

**Review by ayolab — APPROVED** [review]

**Pull request merged**